src/Tools/dotnet-monitor/Egress/Configuration/EgressProviderConfigurationProvider.cs

```diff
@@ -14,19 +16,45 @@ namespace Microsoft.Diagnostics.Tools.Monitor.Egress.Configuration
     internal sealed class EgressProviderConfigurationProvider<TOptions> :
         IEgressProviderConfigurationProvider<TOptions>
     {
-        public EgressProviderConfigurationProvider(IConfiguration configuration, string providerType)
+        private readonly IOptionsTypeToProviderTypesMapper _optionsMapper;
+        private readonly IConfigurationSection _egressSection;
+
+        public EgressProviderConfigurationProvider(IConfiguration configuration, IOptionsTypeToProviderTypesMapper optionsMapper)
         {
-            ProviderType = providerType;
-            Configuration = configuration.GetEgressSection().GetSection(providerType);
+            _optionsMapper  = optionsMapper;
+            _egressSection = configuration.GetEgressSection();
         }
 
         /// <inheritdoc/>
-        public IConfiguration Configuration { get; }
+        public Type OptionsType => typeof(TOptions);
```

Comment:
Since we've unified the Options type and now reference all the categories here, does it even make sense to have this generic parameter anymore?

Reply:
Ah I see these can still be the old azure/filesystem types as well. Can you add comments to clarify this?

Reply:
I need to update all the documentation comments for `IEgressProviderConfigurationProvider`.. let me go do that now.